### PR TITLE
ajustes na tabela de extrato

### LINKED ISSUE #1181: ajustes na tabela de extrato
### Contexto
Percebemos que o extrato da pilot está mostrando algumas linhas de forma estranha ou incorreta. 
Essa issue deve reunir todos os casos ja levantados e possiveis casos para investigação.

#### coluna id da transação
acredito que é interessante adicionar a coluna id da transação quando disponivel para facilitar o usuário conferir a informação do extrato com a tela de transações

#### numero da parcela
atualmente não exibimos o numero da parcela em nenhum caso
Screen Shot 2019-04-19 at 18 48 07

#### extorno de boleto
No estorno de boleto são criadas duas `balance operations` na pilot elas não estão aparecendo corretamente. Na imagem abaixo a primeir linha é referente a `balance operation` de taxa de extorno de boleto. Os dados estão sendo capturados de forma errada no cockpit
Screen Shot 2019-04-18 at 17 46 50
 
Para exemplo como é exibido na dashboard atual, as duas primeiras linhas representam o mesmo extorno de boleto
Screen Shot 2019-04-18 at 17 49 45

#### coluna descrição para boletos
Nas transações de boleto estão apresentando a descrição `cartão de credito`, isso ocorre pois o campo utilizado é `movemen_object.type` quando deveria ser `movement_type.payment_method`
Screen Shot 2019-04-18 at 17 47 02

#### saque está sendo informado como verde e adicionando no calculo
na pilot o saque está adicionando no calculo da tabela ao inves de subtrair
Screen Shot 2019-04-18 at 17 47 17

na dashboard atual é mostrado em laranja e o valor é subtraido
Screen Shot 2019-04-18 at 17 50 07

#### antecipação
aparentemente está correto os dados de antecipação

#### chargeback está sendo exibido errado
chargeback está sendo exibido como entrada e o mdr como saida
Screen Shot 2019-04-20 at 02 16 12

### PR DESCRIPTION
**Pull request by rsmelo**
## Contexto
Este pull request corrige alguns detalhes nas linhas da tabela de operações na tela de extrato

## Issues linkadas
- Resolves #1181 

#### coluna id da transação
adiciona a coluna `id da transação`
Screen Shot 2019-04-22 at 15 47 27

#### numero da parcela
antes
Screen Shot 2019-04-19 at 18 48 07
agora
Screen Shot 2019-04-22 at 15 47 27

#### estorno de boleto
antes
Screen Shot 2019-04-18 at 17 46 50
agora
Screen Shot 2019-04-22 at 15 50 15

#### coluna descrição para boletos
Antes transações de boleto erão apresentando a descrição `cartão de credito`
Screen Shot 2019-04-18 at 17 47 02
agora
Screen Shot 2019-04-22 at 15 52 51

#### saque estava sendo informado como verde e adicionando saldo ao invés de subtrair
Screen Shot 2019-04-18 at 17 47 17
agora
Screen Shot 2019-04-22 at 15 54 44

#### adicionada label `taxa de antecipação`  nos valores de antecipação
antes
Screen Shot 2019-04-22 at 15 58 01
agora
Screen Shot 2019-04-22 at 15 59 09

#### chargeback está sendo exibido errado
antes
Screen Shot 2019-04-22 at 16 00 53

depois
Screen Shot 2019-04-22 at 16 00 31

#### talvez seja necessário marretar alguns status que não são possiveis reproduzir em teste, ou se for mais rápido posso criar um usuário na minha company para quem for testar

- baixar o branch `fix/balance-operations`
- buildar o cockpit
- iniciar a pilot 
- verificar os casos citados nas screenshots acima na tela de extrato

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 004ba6fe50f45ecd2cd0b57d70c6e751f1a05045
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 004ba6fe50f45ecd2cd0b57d70c6e751f1a05045
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 004ba6fe50f45ecd2cd0b57d70c6e751f1a05045
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 004ba6fe50f45ecd2cd0b57d70c6e751f1a05045
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4129e9ac64059be4313aa1768c5d8c514cad3d63
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4129e9ac64059be4313aa1768c5d8c514cad3d63
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4129e9ac64059be4313aa1768c5d8c514cad3d63
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4129e9ac64059be4313aa1768c5d8c514cad3d63
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4129e9ac64059be4313aa1768c5d8c514cad3d63
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4129e9ac64059be4313aa1768c5d8c514cad3d63
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4129e9ac64059be4313aa1768c5d8c514cad3d63
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4129e9ac64059be4313aa1768c5d8c514cad3d63
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

File: packages/cockpit/package.json [hunk 1]
```diff
@@ -20,14 +20,17 @@
     "jest": "24.1.0",
     "jest-cli": "24.1.0",
     "jest-diff": "24.0.0",
+    "moment": "2.24.0",
```
Comment (Deivis) [inline packages/cockpit/package.json]: Ué isso não estava no PR https://github.com/pagarme/pilot/pull/1180 ?

Comment (rsmelo) [reply packages/cockpit/package.json]: o rebase tava zoado, arrumei agora

File: packages/cockpit/src/balance/data/result.js [hunk 2]
```diff
@@ -82,39 +89,137 @@ const getOperationDate = (dateType, fallbackDateType) => pipe(
...
+)
+
+const transferOutcoming = always([
+  {
+    amount: 0,
+    type: 'payable',
+  },
+])
+
+const transferOutgoing = juxt([
```
[the remark this answers is not in the document]
Comment (rsmelo) [reply packages/cockpit/src/balance/data/result.js]: na verdade nesse caso está correto pois no saque são duas saida, uma do valor do saque e outra da taxa de ted

File: packages/pilot/public/locales/pt/translations.json [hunk 3]
```diff
@@ -369,13 +371,16 @@
...
+      "installment": "Parcela",
       "mdr": "MDR",
       "net": "total",
       "outcoming": "entradas",
       "outgoing": "saídas",
       "payment_date": "data",
       "refund": "Estorno",
+      "transaction_id": "ID da Transação",
       "transfer": "Saque",
+      "transfer_fee": "Taxa de transferência",
```
[the remark this answers is not in the document]
Comment (rsmelo) [reply packages/pilot/public/locales/pt/translations.json]: mudei para `TED` como sugerido, tambem alterei `taxa de antecipação` para `Antecipação`

File: packages/cockpit/src/balance/data/result.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (vagnervst) [reply packages/cockpit/src/balance/data/result.js]: Boa, faz sentido!

File: packages/cockpit/src/balance/data/result.js [hunk 4]
```diff
@@ -82,39 +89,137 @@ const getOperationDate = (dateType, fallbackDateType) => pipe(
...
+  ],
+  [
+    pathEq(['movement_object', 'type'], 'credit'),
+    path(['movement_object', 'payment_method']),
+  ],
+  [T, path(['movement_object', 'type'])],
+])
+
+const getInstallment = ifElse(
+  pathEq(['movement_object', 'payment_method'], 'boleto'),
```
Comment (vagnervst) [inline packages/cockpit/src/balance/data/result.js]: Para boleto o `installment` é sempre nulo, acho que pode manter só o `pathOr` neste caso

File: packages/pilot/src/components/Operations/operationsTableColumns.js [hunk 5]
```diff
@@ -102,12 +111,22 @@ const isInvalidAmount = pipe(prop('amount'), either(isNil, Number.isNaN))
 
 const hasInvalidAmount = any(isInvalidAmount)
 
+const buildOperationAmountsReturn = when(
```
Comment (vagnervst) [inline packages/pilot/src/components/Operations/operationsTableColumns.js]: Essa função poderia ter um nome melhor, como `buildEmptyOperationsAmount`

File: packages/cockpit/src/balance/data/result.js (hunk 4 above)
Comment (rsmelo) [reply packages/cockpit/src/balance/data/result.js]: esse caso é especifico para estorno de boleto que acaba botando `installment: 1` no `movement_object`

```JSON
   {
      "object":"balance_operation",
      "id":8176356,
      "status":"available",
      "balance_amount":0,
      "balance_old_amount":null,
      "type":"payable",
      "amount":-1000,
      "fee":0,
      "date_created":"2019-04-17T21:50:54.326Z",
      "movement_object":{
         "object":"payable",
         "id":6817279,
         "status":"paid",
         "amount":-1000,
         "fee":0,
         "anticipation_fee":0,
         "fraud_coverage_fee":0,
         "installment":1,
         "transaction_id":6232675,
         "split_rule_id":null,
         "bulk_anticipation_id":null,
         "recipient_id":"re_cja5ack8i02fkzr6e6l0d3qll",
         "payment_date":"2019-04-17T03:00:00.000Z",
         "original_payment_date":null,
         "type":"refund",
         "payment_method":"boleto",
         "accrual_date":null,
         "date_created":"2019-04-17T21:50:54.319Z"
      }
   }
```

File: packages/pilot/src/components/Operations/operationsTableColumns.js (hunk 5 above)
Comment (rsmelo) [reply packages/pilot/src/components/Operations/operationsTableColumns.js]: é que nesse caso ele não faz o build só das empty como das que não são empty tabem, já que caso não passe no condicional o valor de entrada é retornado. Acho que `buildOperationsAmount` faz sentido, o que acha?

File: packages/pilot/public/locales/pt/translations.json [hunk 6]
```diff
@@ -369,13 +371,16 @@
       "export": "Exportar tudo",
       "gateway_service": "Serviços de gateway",
       "id": "id",
```
Comment (miguelssrs) [inline packages/pilot/public/locales/pt/translations.json]: acho legal colocar a descrição como "ID da Operação" ao invés de somente "ID", porque pode ser confundido 😕

Comment (rsmelo) [reply packages/pilot/public/locales/pt/translations.json]: faz sentido, vou mudar

File: packages/cockpit/src/balance/data/result.js (hunk 4 above)
Comment (vagnervst) [reply packages/cockpit/src/balance/data/result.js]: Puts entendi! Justo manter assim então

File: packages/pilot/src/components/Operations/operationsTableColumns.js (hunk 5 above)
Comment (vagnervst) [reply packages/pilot/src/components/Operations/operationsTableColumns.js]: `buildOperationsAmount` faz sentido

File: packages/pilot/public/locales/pt/translations.json (hunk 6 above)
Comment (rsmelo) [reply packages/pilot/public/locales/pt/translations.json]: corrigido

**Pull request merged**